File: 0733-flood-fill/0733-flood-fill.java
class Solution {
    public int[][] floodFill(int[][] i, int sr, int sc, int nco) {
        if(i[sr][sc]!=nco){
            dfs(i,sr,sc,i[sr][sc],nco);
        }
        return i;
    }
    public void dfs(int[][] i, int r,int c,int co, int nco){
        if(i[r][c]==co){
            i[r][c]=nco;
            if(r>=1) dfs(i,r-1,c,co,nco);
            if(c>=1) dfs(i,r,c-1,co,nco);
            if(r<i.length-1) dfs(i,r+1,c,co,nco);
            if(c<i[0].length-1) dfs(i,r,c+1,co,nco);
        }
        
    }
    
}

File: 0733-flood-fill/0733-flood-fill.py
class Solution:
    def floodFill(self, i: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        
        def tra(r,c, no,co):
            if i[r][c]==no  :
                i[r][c]=co
                if(r>=1): tra(r-1,c,no,co)
                if(c>=1): tra(r,c-1,no,co)
                if(r<len(i)-1): tra(r+1,c,no,co)
                if(c<len(i[0])-1): tra(r,c+1,no,co)
        
        if(i[sr][sc]!=color):
            tra(sr,sc,i[sr][sc],color)
        
        return i